ddht/v5_1/client.py

```diff
@@ -396,6 +399,82 @@ async def find_nodes(
 
                 return responses
 
+    @asynccontextmanager
+    async def stream_find_nodes(
+        self,
+        node_id: NodeID,
+        endpoint: Endpoint,
+        distances: Collection[int],
+        *,
+        request_id: Optional[bytes] = None,
+    ) -> AsyncIterator[trio.abc.ReceiveChannel[InboundMessage[FoundNodesMessage]]]:
+        with self.request_tracker.reserve_request_id(node_id, request_id) as request_id:
+            request = AnyOutboundMessage(
+                FindNodeMessage(request_id, tuple(distances)), endpoint, node_id,
+            )
+
+            send_channel, receive_channel = trio.open_memory_channel[
+                InboundMessage[FoundNodesMessage]
+            ](256)
+            async with trio.open_nursery() as nursery:
+                nursery.start_soon(
+                    self._stream_find_nodes_response,
+                    request,
+                    request_id,
+                    FoundNodesMessage,
+                    send_channel,
+                )
+
+                try:
+                    async with receive_channel:
+                        try:
+                            yield receive_channel
+                        except trio.EndOfChannel as err:
+                            raise trio.TooSlowError from err
+                finally:
```

Comment:
I'm not sure, but I think that this `try/finally` isn't needed.

1. If an exception occurs within the context of this function, it will trigger cancellation of the nursery.
2. If the context exits naturally, this will work just fine without the `try/finally`.